homeassistant/components/awair/config_flow.py

```diff
@@ -21,20 +23,79 @@ class AwairFlowHandler(ConfigFlow, domain=DOMAIN):
 
     VERSION = 1
 
+    _device: AwairLocalDevice
+
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> FlowResult:
+        """Handle zeroconf discovery."""
+
+        host = discovery_info.host
+        LOGGER.debug("Discovered device: %s", host)
+
+        self._device, _ = await self._check_local_connection(host)
+
+        if self._device is not None:
+            await self.async_set_unique_id(self._device.mac_address)
+            self._abort_if_unique_id_configured(updates={CONF_HOST: host})
+            self.context.update(
+                {
+                    "title_placeholders": {
+                        "host": host,
+                        "model": self._device.model,
+                    }
+                }
+            )
+        else:
+            return self.async_abort(reason="unreachable")
+        return await self.async_step_discovery_confirm()
+
+    async def async_step_discovery_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm discovery."""
+        if user_input is not None:
+            title = f"{self._device.model} ({self._device.device_addr})"
+            return self.async_create_entry(
+                title=title,
+                data={CONF_HOST: self._device.device_addr},
+            )
+
+        self._set_confirm_only()
+        placeholders = {"host": self._device.device_addr, "model": self._device.model}
+        return self.async_show_form(
+            step_id="discovery_confirm",
+            description_placeholders=placeholders,
+        )
+
     async def async_step_user(
         self, user_input: dict[str, str] | None = None
     ) -> FlowResult:
         """Handle a flow initialized by the user."""
+
+        if user_input is not None:
+            if CONF_ACCESS_TOKEN in user_input:
+                return await self.async_step_cloud(user_input)
+            if CONF_HOST in user_input:
+                return await self.async_step_local(user_input)
+
```

Comment:
This code is never reached. A menu will automatically redirect to the next step based on the picked option.

```suggestion
```